testdata/errors.go

```diff
@@ -31,6 +31,12 @@ func g(x int) error {
 		return fmt.Errorf("This %d is too low", x) // MATCH /error strings.*not be capitalized/
 	} else if x == 0 {
 		return fmt.Errorf("XML time") // ok
+	} else if x > 3 {
+		return fmt.Errorf("newlines are fun\n") // MATCH /error strings.*not end with a newline/
+	} else if x > 2 {
+		return fmt.Errorf("Newlines are really fun\n") // MATCH /error strings.*or end with a newline/
+	} else if x > 1 {
+		return fmt.Errorf("newlines are\nespeciallyfun\non a console\n")
 	}
 	return errors.New(`too much stuff.`) // MATCH /error strings.*not end with punctuation/
```

Comment:
How about we drop that whole if/switch business and just have a bunch of consecutive returns? golint doesn't care that the statements are unreachable, so I see no point to make it look like sensible code. Something like this should do just fine:

```
func g(x int) error {
	return fmt.Errorf("This %d is too low", x)     // MATCH /error strings.*be capitalized/
	return fmt.Errorf("XML time")                  // ok
	return fmt.Errorf("newlines are fun\n")        // MATCH /error strings.*end with punctuation/
	return fmt.Errorf("Newlines are really fun\n") // MATCH /error strings.+not be capitalized/
	return errors.New(`too much stuff.`)           // MATCH /error strings.*end with punctuation/
}
```